src/frontend/js/components/PaymentButton/components/PaymentInterfaces/LyraPopIn.tsx

```diff
@@ -0,0 +1,121 @@
+
+const LyraPopIn = ({
+  onSuccess,
+  onError,
+  configuration,
+  form_token,
+}: Joanie.PaymentInterfaceProps<Joanie.LyraPayment>) => {
+  const intl = useIntl();
+  const shouldAbort = useRef<Boolean>(true);
+
+  const handleError = (error?: Error) => {
+    if (error) handle(`[LyraPopIn] - ${error}`);
+
+    if (shouldAbort) {
+      onError(PaymentErrorMessageId.ERROR_ABORTING);
+    }
+    onError(PaymentErrorMessageId.ERROR_DEFAULT);
+  };
+
+  const initializeLyraForm = async () => {
+    const { base_url: baseUrl, public_key: publicKey } = configuration;
+
+    const link = document.createElement('link');
+    link.setAttribute('rel', 'stylesheet');
+    link.setAttribute('href', `${baseUrl}/static/js/krypton-client/V4.0/ext/neon-reset.min.css`);
+    document.head.appendChild(link);
+
+    const script = document.createElement('script');
+    script.src = `${baseUrl}/static/js/krypton-client/V4.0/ext/neon.js`;
+    document.head.appendChild(script);
+
+    const { KR } = await KRGlue.loadLibrary(baseUrl, publicKey).catch((error) => {
+      handleError(error);
+      return { KR: null };
+    });
+
+    return KR;
+  };
+
+  useAsyncEffect(async () => {
+    const KR = window.KR ?? (await initializeLyraForm());
+    if (KR === null) return;
+
+    await KR.setFormConfig({
+      formToken: form_token,
+      'kr-language': intl.locale,
+      'kr-spa-mode': true,
+      'kr-z-index': '400',
+    });
+
+    // Create a container for the lyra form which must be out of the React tree.
+    // This is necessary to avoid the form being unmounted when the component is.
+    const formContainer = document.createElement('div');
+    formContainer.id = LYRA_FORM_ID;
+    formContainer.classList.add('kr-embedded');
+    formContainer.setAttribute('kr-popin', '');
+    formContainer.style.visibility = 'hidden';
+    document.body.appendChild(formContainer);
+
+    const {
+      result: { formId },
+    } = await KR.renderElements('#lyra-form');
+
+    const handleFormSubmit = async (response: KRPaymentResponse) => {
+      shouldAbort.current = false;
+      await KR.closePopin(formId);
+      if (response.clientAnswer.orderStatus === 'PAID') {
+        onSuccess();
+        return true;
+      }
+
+      handleError();
+      return false;
+    };
+
+    const handleFormError = async (error: KRError) => {
+      // Do not close the pop-in if the error is a client error.
+      if (!error.errorCode.startsWith('CLIENT_')) {
+        shouldAbort.current = false;
+        await KR.closePopin(formId);
+        handleError();
+      }
+    };
+
+    const handleClosePopIn = () => {
+      if (shouldAbort.current === true) {
+        onError(PaymentErrorMessageId.ERROR_ABORTING);
+      }
+    };
+
+    const openPopIn = async () => {
+      await KR.openPopin(formId);
+    };
+
+    await KR.onSubmit(handleFormSubmit);
+    await KR.onError(handleFormError);
+    await KR.onPopinClosed(handleClosePopIn);
+    await KR.onFormReady(openPopIn);
```

Comment:
Unusual pattern here, 4 `await` 🤔, interesting

Reply:
Better with 
```ts
  await Promise.all([
      KR.onSubmit(handleFormSubmit),
      KR.onError(handleFormError),
      KR.onPopinClosed(handleClosePopIn),
      KR.onFormReady(openPopIn),
    ])
```
?

Reply:
I'll check what happend if I do not use await. To setup those callbacks I don't think I have to wait that each of them has resolved

Reply:
Indeed, after a quick test, it's useless to await those methods.